tests/decoding/test_intervals_removal.py

```diff
@@ -0,0 +1,299 @@
+"""Tests for intervals dimension removal.
```

Comment:
With the file name `test_interval_removal` make sense to us down the road? Maybe `test_interval_structure.py`?